sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -2356,6 +2358,68 @@ test_suite(
     tests = ["epmp_init_otp_" + lc_state for lc_state, _ in get_lc_items()],
 )
 
+platform_target(
```

Comment:
Two things on this:
1. I remember there was a reason we couldn't apply the transition to the cc_binary rule in opentitan.bzl, but I can't remember what it was now, did you encounter why this was? Because if we could, then we wouldn't need this wrapper here (i.e., we could apply the transition directly on the cc_binary rule in the `opentitan_binary` macro.
2. If you do need this transition, then I think a better place for this would be in `sw/device/silicon_creator/rom/BUILD`, then you can just use it in this BUILD file.

Reply:
Sorry, I don't speak Bazel well enough to know the answer. @pamaury

Reply:
no problem, I believe it is an orthogonal issue, so perhaps we can just do item 2 above, and rebase this PR on master so we can get it merged. wdyt?

Reply:
Moved to rom/BUILD and rebased. PTAL

Reply:
Thanks! LGTM.

Reply:
I think the proper solution is to change `opentitan_binary` but since this was a potentially dangerous change in the last push before the deadline, we decided to just add a manual transition. I can look into changing `opentitan_binary` if that's alright?

Reply:
yep that sounds good, I think thats the solution we were looking for it just never made its way into the repo hehe